nidirect_contacts/src/Controller/ContactListingController.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Drupal\nidirect_contacts\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\views\Views;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class ContactListingController.
+ */
+class ContactListingController extends ControllerBase {
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Drupal\Core\Block\BlockManagerInterface definition.
+   *
+   * @var \Drupal\Core\Block\BlockManagerInterface
+   */
+  protected $blockManager;
+
+  /**
+   * Symfony\Component\HttpFoundation\RequestStack definition.
+   *
+   * @var Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $request;
+
+  /**
+   * Constructs a new ContactListingController object.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, BlockManagerInterface $block_manager, RequestStack $request) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->blockManager = $block_manager;
+    $this->request = $request;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('plugin.manager.block'),
+      $container->get('request_stack')
+    );
+  }
+
+  /**
+   * Default presentation of the contacts listing page.
+   *
+   * - Title (Contacts A to Z)
+   * - Search by contact name
+   *    <input 'Search contacts' placeholder> <submit>
+   * - Find contacts beginning with...
+   *    <Big A-Z, 0-9 grid of links>
+   *    Link is: /contacts/letter/a
+   *
+   * @return array
+   *   Return render array for Drupal to convert to HTML.
+   */
+  public function default() {
+    $content = [];
+
+    // Views static service class used in absence of container service to inject via constructor.
```

Comment:
Not sure how this comment is relevant to the render array?

Reply:
Me neither, and I wrote it! Putting it down to a brain-fart and slimming down the comment block with the next commit :)